src/engine/platform/mixer.cpp

```diff
@@ -47,6 +55,53 @@ struct AudioStruct {
     SDL_AudioDeviceID dev;
 } audio_struct = {};
 
+Effect create_delay(f32 feedback, f32 delay_time) {
+    ASSERT(delay_time < CHANNEL_BUFFER_LENGTH_SECONDS, "Delay time is longer than channel buffer");
+    auto delay_func = [] (Effect *effect, f32 *buffer, u32 start, u32 len) -> void {
+        if (effect->delay._delay_len_seconds != effect->delay._prev_delay_len_seconds) {
+            //TODO(GS) crackling when length is changed while sound is playing
+            effect->delay.delay_len = (u32) AUDIO_SAMPLE_RATE * effect->delay._delay_len_seconds * 2;
+            effect->delay._prev_delay_len_seconds = effect->delay._delay_len_seconds;
+        }
+        for (u32 i = 0; i < len; i++) {
+            //                                                    vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+            buffer[(start + i) % CHANNEL_BUFFER_LENGTH] += buffer[ABS(((s32) start + i - effect->delay.delay_len) % CHANNEL_BUFFER_LENGTH)] * effect->delay.feedback;
+        }
+    };
+    Effect effect = {delay_func};
```

Comment:
Maybe break this out to assign to the field directly, if you can't initalize everything in the initalizer-list-thingy I would rather see setting of each field.